New SVG diagram guidance (#46)
openownership/bods-dev-handbook: diagram_creation.md

# Creating SVG diagrams

> BODS diagrams are created in Scalable Vector Graphic (SVG) format for ease of scaling and translation. SVGs use XML syntax. Text can therefore be stripped out and translated using standard tools. The following workflow guidance is provided so that diagrams within the BODS documentation can accommodate translated text.

Reviewer: ```suggestion
BODS diagrams are created in [Scalable Vector Graphic (SVG) format](https://www.w3.org/TR/SVG2/) for ease of scaling and translation. SVGs use XML syntax. Text can therefore be stripped out and translated using a wide range of text and graphics editors. The following workflow guidance is provided so that diagrams within the BODS documentation can accommodate translated text.
```

Author: I edited along these lines.

## Suggested diagram creation workflow

> - Use Google Draw to do a 1st draft of a diagram BUT do not include any text.

Reviewer: ```suggestion
- Use Google Draw to do a first draft of a diagram BUT do not include any text.
```

  This is recommended since Google Draw allows for easy layout, styling and connection of diagram elements. Examples of Google Draw versions of diagrams can be [seen here](https://drive.google.com/drive/folders/1R0r-Dx-PivOYmUPbE7MtKbHm-nrFAikB), though note that they have text in them.
  
- Export the diagram as an SVG file using File > Download.

- Open the SVG locally in [Inkscape >1.0](https://inkscape.org/).

  Since verion 1, Inkscape has [better SVG 2 support and text-handling](https://wiki.inkscape.org/wiki/index.php?title=Release_notes/1.0#Browser-compatible_flowed_text). 

- Go to Edit > Preferences > Text and ensure that 'Use SVG2 auto-flowed text' is checked.

>   The [SVG 2 specification](https://www.w3.org/TR/2018/CR-SVG2-20181004/) contains support for text-wrapping, and Inkscape produces good SVG 2 output. However browsers still do not implement SVG 2 fully. So the guidance below contains workarounds.

Reviewer: ```suggestion
  The [SVG 2 specification](https://www.w3.org/TR/2018/CR-SVG2-20181004/) contains support for text-wrapping, and Inkscape produces good SVG 2 output. However browsers still do not implement SVG 2 fully, so the guidance below contains workarounds.
```
